addons/main/functions/GroupAction/fnc_doGroupAssault.sqf

```diff
@@ -20,30 +20,34 @@ params ["_cycle", "_units", "_pos"];
 
 // update
 _units = _units select {_x call FUNC(isAlive) && {!isPlayer _x}};
-if (_units isEqualTo []) exitWith {};
+if (_units isEqualTo [] || {_pos isEqualTo []}) exitWith {false};
 
+private _targetPos = _pos deleteAt 0;
 {
-    private _targetPos = selectRandom _pos;
-    // setpos
-    if (RND(0.5) || {(currentCommand _x) isNotEqualTo "MOVE"}) then {
-        _x doMove _targetPos;
-        _x setDestination [_targetPos, "FORMATION PLANNED", false]; // added to reduce cover bounding - nkenny
-        _x lookAt _targetPos;
-    };
-
     // manoeuvre
-    _x forceSpeed 4;
+    _x forceSpeed 3;
     _x setUnitPosWeak (["UP", "MIDDLE"] select (getSuppression _x isNotEqualTo 0));
     _x setVariable [QGVAR(currentTask), "Group Assault", GVAR(debug_functions)];
     _x setVariable [QEGVAR(danger,forceMove), true];
+
+    // check enemy
+    private _enemy = _x findNearestEnemy _x;
```

Comment:
ne need to assign a variable

Reply:
Would 
`
if (_x distance2D (_x findNearestEnemy _x) < 12) then {_targetPos = getPosATL (_x findNearestEnemy _x);};
`
Be better.  Or am I misunderstanding something?

Reply:
it is readability vs speed in this case.

I would prefer speed here as assigning a variable that is only used once seems a bit of a waste to me.

Reply:
kk

Reply:
Fixed